src/Encoder/BlockCipher.js

```diff
@@ -0,0 +1,328 @@
+// import md5 from './Hash/md5'
+import nodeCrypto from 'crypto'
+
+const meta = {
+  name: 'block-cipher',
+  title: 'Block Cipher',
+  category: 'Modern cryptography',
+  type: 'encoder'
+}
+
+const algorithms = [
+  {
+    name: 'aes-128',
+    label: 'AES-128',
+    blockSize: 128,
+    keySize: 128,
+    browserAlgorithm: 'aes',
+    nodeAlgorithm: 'aes-128'
+  },
+  {
+    name: 'aes-192',
+    label: 'AES-192',
+    blockSize: 128,
+    keySize: 192,
+    browserAlgorithm: 'aes',
+    nodeAlgorithm: 'aes-192'
+  },
+  {
+    name: 'aes-256',
+    label: 'AES-256',
+    blockSize: 128,
+    keySize: 256,
+    browserAlgorithm: 'aes',
+    nodeAlgorithm: 'aes-256'
+  }
+
+]
+
+const modes = [
+  {
+    name: 'ecb',
+    label: 'ECB (Electronic Code Book)',
+    needsIV: false
+  },
+  {
+    name: 'cbc',
+    label: 'CBC (Cipher Block Chaining)',
+    needsIV: true
+  },
+  {
+    name: 'ctr',
+    label: 'CTR (Counter)',
+    needsIV: true
+  }
+]
+
+/**
+ * Encoder brick for creating message digests
+ */
+export default class BlockCipherEncoder extends Encoder {
+  /**
+   * Returns brick meta.
+   * @return {object}
+   */
+  static getMeta () {
+    return meta
+  }
+
+  /**
+   * Constructor
+   */
+  constructor () {
+    super()
+
+    const algorithms = BlockCipherEncoder.filterAvailableAlgorithms()
+    this.registerSetting([
+      {
+        name: 'algorithm',
+        type: 'enum',
+        value: 'aes-128',
+        randomizable: false,
+        options: {
+          elements: algorithms.map(algorithm => algorithm.name),
+          labels: algorithms.map(algorithm => algorithm.label)
+        }
+      },
+      {
+        name: 'mode',
+        type: 'enum',
+        value: 'cbc',
+        randomizable: false,
+        options: {
+          elements: modes.map(algorithm => algorithm.name),
+          labels: modes.map(algorithm => algorithm.label)
+        }
+      },
+      {
+        name: 'key',
+        type: 'bytes',
+        value: new Uint8Array([
+          0x2b, 0x7e, 0x15, 0x16, 0x28, 0xae, 0xd2, 0xa6,
+          0xab, 0xf7, 0x15, 0x88, 0x09, 0xcf, 0x4f, 0x3c ]
+        )
+      },
+      {
+        name: 'iv',
+        type: 'bytes',
+        value: new Uint8Array([
+          0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+          0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F ]
+        )
+      } /*,
+      {
+        name: 'pad',
+        type: 'boolean',
+        value: false
+      } */
+
+    ])
+  }
+
+  /**
+   * Triggered when a setting value has changed.
+   * Override is required to call super.
+   * @protected
+   * @param {Setting} setting
+   * @param {mixed} value Setting value
+   */
+  settingValueDidChange (setting, value) {
+    super.settingValueDidChange(setting, value)
+
+    if (setting.getName() === 'mode') {
+      let ivSetting = this.getSetting('iv')
+      const isVisible = (value !== 'ecb')
+      ivSetting && ivSetting.setVisible(isVisible)
```

Comment:
We can depend on settings registered in the constructor. No need for additional checks.

```suggestion
      this.getSetting('iv').setVisible(value !== 'ecb')
```